### ISSUE TITLE
Coala verbose mode

### ISSUE THREAD

**Issue by anctartica** [coala/coala#5659]
If coala verbose setting is used then if a linter bear also has verbose mode it should also be used.

**Comment by RaiVaibhav** [comment]
> this fits into debugging . Currently we dont see verbose output from linters. We need to in order to debug problems in the linter.

@jayvdb  I need help in understanding this, its not clear to me what you want debug,
@anctartica  every bear have `section` variable in there `__init__` you can add logic if you want tocheck whether verbose mode is on in coala or not 
```
if self.section['log_level'].value is 'DEBUG:
    args = ('--verbose', filename,)
```

**Comment by RaiVaibhav** [comment]
And same can be done for any bear

**Comment by jayvdb** [comment]
Nice. Do we have any tests that confirm logging level can be different per section in a .coafile?

Do we have any validation on this setting? What if the .coafile setting uses 'debug' instead of 'DEBUG'?

If the section log level has debug enabled, and the linter emits something which doesnt match the regexes, can it be assigned to debug level?  We dont want to try to regex match every possible debug msg emitted by a linter - we exclude one and the log could be very confusing.

**Comment by RaiVaibhav** [comment]
>Do we have any tests that confirm logging level can be different per section in a .coafile?

No `log_level` don't even support `.coafile` if you even set `log_level` to `DEBUG` or anything it will not go into `verbose mode`, these will convert into section `setting`  and we can use that, I thought it was for **CLI** part only, as label also have `area/CLI` the above comment is only for CLI part.

**Comment by jayvdb** [comment]
Well, typically a user/dev wants to debug one bear, or one .coafile section, and debugging all sections in verbose mode will blow the CI log size maximum causing job to be terminated prematurely, preventing possibly preventing debugging of the problematic section.

Can we add per section log level?  A low issue or medium?

**Comment by RaiVaibhav** [comment]
>Can we add per section log level? A low issue or medium?

Yeah we can But I have to talk with either @Makman2  or some maintainer who is involved in core right now, as currently, I don't know why explicitly `log_level` settings not given to a user in `.coafile`

**Comment by RaiVaibhav** [comment]
>debugging all sections in verbose mode will blow the CI log

Doesn't this is currently coala doing right now? and here `debugging` word is still confusing as debugging per bear (debug_bears) is good right now

**Comment by RaiVaibhav** [comment]
`log_level` per bear is tough and `log_level` per section is ok but one thing here according to my solution all the logging before [this line](https://github.com/coala/coala/blob/master/coalib/coala_main.py#L162) will not be log  in section level logging, if I put the code 
```
 str_log_level = str(section.get('log_level', '')).upper()
logging.getLogger().setLevel(LOG_LEVEL.str_dict.get(str_log_level,
                                                    LOG_LEVEL.INFO))
```
just after line 162.

And please don't assign the issue if `section` level logging to me :P, I want someone else to do as we have very fewer people in the core and I want someone else to take interest in core part ( also I am still at the bottom in that list.)

**Comment by RaiVaibhav** [comment]
@jayvdb  any update/info/comment ?
